### PR TITLE
Updating package release to run through the ESRP release pipeline

### LINKED ISSUE #28160: Transition release process to ESRP Gateway
Currently, we directly release using `twine upload`.

We need to swap this confirmation to release using the [ESRP release task](https://github.com/Azure/azure-sdk-for-java/blob/d183b5e1ba94d4786d290bc2c4357a95fda47b8a/eng/pipelines/templates/steps/java-publishing.yml#L101).

Also document:

- [x] How to add an owner to a package on PyPI
- [x] The process to YANK or DELETE any given package

### PR DESCRIPTION
**Pull request by scbedd**
Resolves #28160 

This PR enables the usage of `ESRP Release` task for publish to PyPI. SDK Devs don't need to worry about this update, as the release process remains the same.

This PR:

- [x] Updates our release process to use the `ESRP Release` task instead of `twine upload` using our `azure-sdk` account.
- [x] On pypi, these packages will be published by the `Microsoft` account instead of the `azure-sdk` account.
- [x] Not reflected in PR, but I also proved out the update to our [partner drops pipeline](https://dev.azure.com/azure-sdk/internal/_apps/hub/ms.vss-releaseManagement-web.cd-release-progress?_a=release-environment-logs&releaseId=23003&environmentId=28631).

Check `doc/esrp_release.md` for additional detail on break glass scenarios.

### CODE REVIEW

File: doc/esrp_release.md
Comment (benbp) [inline doc/esrp_release.md:7]:
```suggestion
The primary visible effect of these release changes is that these packages will be published by the `Microsoft` pypi account, _not_  `azure-sdk` account that has been used until now.
```

**Review by benbp — APPROVED** [review]

File: doc/esrp_release.md
Comment (scbedd) [reply doc/esrp_release.md:7]: Oh the nitpick of justice.

Comment (benbp) [reply doc/esrp_release.md]: Feel my wrath

**Comment by scbedd** [comment]
I also need to adjust this PR to allow the partner release pipeline to use this same strategy. That commit will be added before merge after all the releases have finished up.

File: eng/pipelines/templates/stages/archetype-python-release.yml
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-python-release.yml:110]: Is a copy actually necessary or can we just pass the existing folder?

File: eng/pipelines/templates/stages/archetype-python-release.yml
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-python-release.yml:123]: I know you just copied this but we should probably change this to a more generic alias either our team or bot name. Or mabye we can see if this is actually required at all.

Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-python-release.yml:123]: The "approvers" field here? Yeah I'll ask Akoly.

File: eng/pipelines/templates/stages/archetype-python-release.yml
Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-python-release.yml:110]: The folder that we pass in gets a new artifact dropped in it. That new artifact it drops is _also_ a zip which breaks the succeeding step.

So we either add a manual step to copy these things off to a different folder, or a manual step to clean up the new artifact that's dropped. Either way I gotta do something.

I don't want to place the publish to integration above the actual release to PyPI, for consistency reasons.

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-python-release.yml:110]: I was just trying to avoid unnecessary copying or deleting and instead just have the uploaded artifact match what we need for publishing without modification when downloaded. I'm sure your remember all the extra filtering we did in the various release steps in the past that is what I'm trying to avoid. However, if that doesn't work with the other ESRP release then copying should be fine.

File: eng/pipelines/templates/stages/archetype-python-release.yml
Comment (scbedd) [reply eng/pipelines/templates/stages/archetype-python-release.yml:123]: Ok looks likeit can be any single email. Going to swap this to the RequestedForEmail.

**Comment by scbedd** [comment]
Commenting old PR body for posterity.

@hallipr @JimSuplizio I basically copy-pasta-ed your configuration from [sdk-for-java](https://github.com/Azure/azure-sdk-for-java/blob/bff0243077172420ae250b25c9e3001734140069/eng/pipelines/templates/steps/java-publishing.yml#L100).

Excepting:

- Targeting `PyPI` instead of `Maven`
- Using Publisher `ESRPRELTEST` instead of `ESRPRELPACMANTEST` as that is what was used in the [demo](https://microsoftapc-my.sharepoint.com/:v:/r/personal/shmallip_microsoft_com/Documents/Recordings/ESRP%20Release%20PyPI%20Publishing%20Demo%20(Client_%20AzureSDK)-20230104_223533-Meeting%20Recording.mp4?csf=1&web=1&e=WquPa8) provided to me.

Any gotchas that I should be aware of other than actually running this thing?

EDIT: according to [filed icm](https://portal.microsofticm.com/imp/v3/incidents/details/371323774/home) the ESRP team does not currently support uploading a `tar.gz` formatted sdist. PR is pending until they support it.

Further EDIT: 3/6/2023 ESRP team got back to me, re-ran the release, zip sdist is present now.

image

**Pull request merged**
